runnable/extra-files/dexcept.h

```diff
@@ -0,0 +1,63 @@
+// D exceptions to by caught by C++
+
+#pragma once
+
+namespace D {
+
+struct string
+{
+    size_t length;
+    const char* ptr;
+};
+
+class Object;
+
+extern "C" {
+    string _d_toString(Object* o);
+}
+
+class Object
+{
+    void* __monitor;
+
+    virtual void _classinfo_data() = 0;
+
+protected:
+    // don't call directly, ABI mismatch

```

Comment:
Could you elaborate a bit on the ABI mismatch? I assumed ABI compliance between D and C++ would already be checked by dmd-testsuite...